pkg/providers/instancetype/suite_test.go

```diff
@@ -1186,6 +1185,22 @@ var _ = Describe("InstanceType Provider", func() {
 			vm := azureEnvNonZonal.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Pop().VM
 			Expect(vm.Zones).To(BeEmpty())
 		})
+		It("should support provisioning non-zonal instance types in zonal regions", func() {
+			nonZonalInstanceType := "Standard_NC6s_v3"
```

Comment:
Is it mainly GPUs that are nonzonal? Or just skus with limited capacity? I am so curious about how all of this works

Reply:
Not sure yet. Will look further into why certain SKUs are non-zonal.